examples/v1beta1/pipelineruns/pipelinerun-with-when-expressions.yaml

```diff
@@ -0,0 +1,239 @@
+
+        if [[ "$(params.deleteExisting)" == "true" ]] ; then
+          cleandir
+        fi
+
+        test -z "$(params.httpProxy)" || export HTTP_PROXY=$(params.httpProxy)
+        test -z "$(params.httpsProxy)" || export HTTPS_PROXY=$(params.httpsProxy)
+        test -z "$(params.noProxy)" || export NO_PROXY=$(params.noProxy)
+
+        /ko-app/git-init \
+          -url "$(params.url)" \
+          -revision "$(params.revision)" \
+          -refspec "$(params.refspec)" \
+          -path "$CHECKOUT_DIR" \
+          -sslVerify="$(params.sslVerify)" \
+          -submodules="$(params.submodules)" \
+          -depth "$(params.depth)"
+        cd "$CHECKOUT_DIR"
+        RESULT_SHA="$(git rev-parse HEAD | tr -d '\n')"
+        EXIT_CODE="$?"
+        if [ "$EXIT_CODE" != 0 ]
+        then
+          exit $EXIT_CODE
+        fi
+        # Make sure we don't add a trailing newline to the result!
+        echo -n "$RESULT_SHA" > $(results.commit.path)
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: check-file
+spec:
+  params:
+    - name: path
+  workspaces:
+    - name: source
+      description: The workspace to check for the file.
+  results:
+    - name: status
+      description: indicating whether the file exists
+  steps:
+    - name: check-file
+      image: alpine
+      script: |
+        if test -f $(workspaces.source.path)/$(params.path); then
+          printf exists | tee /tekton/results/status
+        else
+          printf missing | tee /tekton/results/status
+        fi
+---
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  name: guarded-pipeline
+spec:
+  params:
+    - name: path
+      type: string
+      description: The path of the file to be created.
+      default: "README.md"
+    - name: repo-url
+      type: string
+      description: The git repository URL to clone from.
+    - name: branch-name
+      type: string
+      description: The git branch to clone.
+  workspaces:
+    - name: source-repo
+      description: |
+        This workspace will receive the cloned git repo and be passed
+        to the next Task to create a file.
+  tasks:
+    - name: fetch-repo
+      taskRef:
+        name: git-clone-from-catalog
+      params:
+        - name: url
+          value: $(params.repo-url)
+        - name: revision
+          value: $(params.branch-name)
+      workspaces:
+        - name: output
+          workspace: source-repo
+    - name: create-file
+      when:
+        - input: "$(params.path)"
+          operator: in
+          values: ["README.md"]
+      workspaces:
+        - name: source
+          workspace: source-repo
+      runAfter:
+        - fetch-repo
+      taskSpec:
+        workspaces:
+          - name: source
+            description: The workspace to create the readme file in.
+        steps:
+          - name: write-new-stuff
+            image: ubuntu
+            script: 'touch $(workspaces.source.path)/README.md'
+    - name: check-file
+      when:
```

Comment:
this `when` is not needed here since its always `true`, we can drop this from here. And instead we could add this in the doc to demonstrate different `when` expressions